app/admin_components/adf-services/dreamfactory-services.js

```diff
@@ -348,6 +348,45 @@ angular.module('dfServices', ['ngRoute', 'dfUtility'])
                         scope.applyFilter();
                     }
                 });
+
+                var goToSelectedApiDoc = function () {
+                    
+                    if (!scope.dataLoading) {
+                        // We want to seperate out the service name, which can be done by pulling out anything
+                        // after the second hash.
+                        var url = top.window.location.hash,
+                        delimiter = '#',
+                        start = 2,
+                        serviceNameAsArray = url.split(delimiter).slice(start),
+                        selectedServiceName = serviceNameAsArray.toString();
+                        
+                        function isCurrentService(service) {
+                            return service.record.name === selectedServiceName;
+                        }
+                        // We then check the selectedServiceName matches to one of the services in the 
+                        // ApiDocs table. If it does, then we open that Api Doc.
+                        var selectedService = scope.filteredServices.find(isCurrentService);
+                        
+                        if (selectedService) {
+                            scope.editService(selectedService.record);
+                        }
+                    } else {
+                        // Wait until all services have loaded in before going to that API Doc.
+                        setTimeout(goToSelectedApiDoc, 300);
```

Comment:
When you go to an API Doc, from the services dashboard, you will notice that all the services load up for a brief period, and then it loads up the individual API Doc for the service in question.
Its not ideal, any thoughts?

Reply:
We should not wait for it to load. Same thing as in df-admin-app PR.  We should subscribe to the success event of the Promise. In this case, the function will only fire when the data is there.

Waiting is bad because maybe the data will never load.

Reply:
So please find where we load all services and put function call there.

Reply:
Good call, thanks, this is done